packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema-v2.ts

```diff
@@ -215,44 +215,33 @@ const buildAPIProject = async (
 };
 
 const _buildProject = async (context: $TSContext, opts: TransformerProjectOptions<TransformerFactoryArgs>): Promise<DeploymentResources> => {
-  const userProjectConfig = opts.projectConfig;
-  const stackMapping = userProjectConfig.config.StackMapping;
-  const userDefinedSlots = {
-    ...parseUserDefinedSlots(userProjectConfig.pipelineFunctions),
-    ...parseUserDefinedSlots(userProjectConfig.resolvers),
-  };
-
-  // Create the transformer instances, we've to make sure we're not reusing them within the same CLI command
-  // because the StackMapping feature already builds the project once.
   const transformers = await opts.transformersFactory(opts.transformersFactoryArgs);
   const transform = new GraphQLTransform({
     transformers,
-    stackMapping,
-    transformConfig: userProjectConfig.config,
+    stackMapping: opts.projectConfig.config.StackMapping,
+    transformConfig: opts.projectConfig.config,
     authConfig: opts.authConfig,
     buildParameters: opts.buildParameters,
     stacks: opts.projectConfig.stacks || {},
     featureFlags: new AmplifyCLIFeatureFlagAdapter(),
     sandboxModeEnabled: opts.sandboxModeEnabled,
-    userDefinedSlots,
+    userDefinedSlots: parseUserDefinedSlotsFromProject(opts.projectConfig),
     resolverConfig: opts.resolverConfig,
```

Comment:
what is the difference between resolverConfig and transformConfig exactly?

Reply:
resolverConfig sets up conflict resolution config on data sources, while transformConfig is just a bag of bits. I'm currently refactoring some of the vague things like transformConfig to be specific input params, so the input to the `constructTransform` is as small, and specific as possible.